src/Umbraco.Core/Events/EventAggregator.Notifications.cs

```diff
@@ -84,17 +86,63 @@ internal abstract class NotificationAsyncHandlerWrapper
     internal class NotificationAsyncHandlerWrapperImpl<TNotification> : NotificationAsyncHandlerWrapper
         where TNotification : INotification
     {
+        /// <remarks>
+        /// <para>
+        /// Background - During v9 build we wanted an in-process message bus to facilitate removal of the old static event handlers. <br/>
+        /// Instead of taking a dependency on MediatR we (the community) implemented our own using MediatR as inspiration.
+        /// </para>
+        ///
+        /// <para>
+        /// Some things worth knowing about MediatR.
+        /// <list type="number">
+        /// <item>All handlers are by default registered with transient lifetime, but can easily depend on services with state.</item>
+        /// <item>Both the Mediatr instance and its handler resolver are registered scoped and as such it is always possible to depend on scoped services in a handler.</item>
+        /// </list>
+        /// </para>
+        ///
+        /// <para>
```

Comment:
I'm shaking my head reading this. I cannot believe it was changed to a singleton.

Reply:
I thought it was a bit odd https://github.com/umbraco/Umbraco-CMS/commit/ea790511040e6cc9d1fc2ed0a7c9428d611577f8

No one is perfect, shit happens, not here to grumble only to work out how to make it better going forwards.

Reply:
I done messed up there though!! Sorry to everyone.